File: php/CancelWaybill.php
<?php
/**
cancel the waybill
*/
include_once "EncryptUtil.php";
include_once "HttpRequestUtil.php";

$requestTime = time(); //timestamp of request
$secretKey = "fbc0ba8250b813275c767390a4bc7d3b"; //secretKey from Fodel
$params = array();
$params["awb"] = "534053190";//awb number
$params["app_key"] = "2003"; //app key from Fodel
$params["ts"] = $requestTime;
//generate the sign from the paramters with eht secretKey
$signString = EncryptUtil::generateSign($params,$secretKey);
$params['sign'] = $signString;

//Set header ,add the api version to the parameter fodel_api_version
$header = array();
$header[] = "fodel_api_version: v1";
$httprequest = new HttpRequestUtil();
$httprequest->setBaseUrl("http://api.test.fo-del.com");//host of fodel
$respone = $httprequest->post("/shipment/cancel",json_encode($params),true,$header);
echo $respone->body;

File: java/CancelWaybill.java

import com.alibaba.fastjson.JSONException;
import com.alibaba.fastjson.JSONObject;

import java.io.IOException;
import java.text.ParseException;
import java.util.HashMap;
import java.util.Map;

public class CancelWaybill {
	public static void main(String[] args) throws JSONException, IOException, ParseException, FodelAPIException{
		String secretKey = "";//secretKey from Fodel
		String requestTime = (System.currentTimeMillis()/1000)+""; //request time
		Map<String ,String> params = new HashMap<String,String>();
		params.put("app_key", "");//app key from fodel
		params.put("awb", "");//awb number
		params.put("ts", requestTime);
		String signString = EncryptUtil.generateSign(params, secretKey);
		params.put("sign", signString);
		JSONObject jsonObject = HttpRequestUtil.request("POST", "http://api.test.fo-del.com/parcel/cancel", params);
		System.out.println(jsonObject);
	}
}
